frontend/app/components/wp-relations/wp-relations.directive.ts

```diff
@@ -26,123 +26,147 @@
 // See doc/COPYRIGHT.rdoc for more details.
 //++
 
-import {wpTabsModule} from '../../angular-modules';
-import {WorkPackageRelationGroup} from './wp-relation-group/wp-relation-group.service';
-import {WorkPackageNotificationService} from '../wp-edit/wp-notification.service';
-import {WorkPackageResource} from '../api/api-v3/hal-resources/work-package-resource.service';
+import {wpDirectivesModule} from "../../angular-modules";
+import {RelatedWorkPackage, RelatedWorkPackagesGroup} from './wp-relations.interfaces';
 
-const iconArrowClasses = ['icon-arrow-up1', 'icon-arrow-down1'];
+import {WorkPackageResourceInterface} from '../api/api-v3/hal-resources/work-package-resource.service';
+import {WorkPackageCacheService} from '../work-packages/work-package-cache.service';
 
 export class WorkPackageRelationsController {
-  public btnTitle:string;
-  public btnIcon:string = '<i class="icon-hierarchy icon-add"></i>';
-
-  public workPackage:WorkPackageResource;
-  public relationGroup:WorkPackageRelationGroup;
-  public focusElementIndex:number = -2;
-  public wpToAddId:number = null;
-  public expand:boolean = false;
-  public text:any;
-
-  private _initialExpand:boolean;
-
-  public get stateClass():string {
-    return iconArrowClasses[+!!this.expand];
+  public relationGroups:RelatedWorkPackagesGroup;
+  public workPackage:WorkPackageResourceInterface;
+
+  public currentRelations: Array<RelatedWorkPackage> = Array();
+  private sortByWpType:boolean;
+
+  constructor(protected $scope:ng.IScope,
+              protected $q:ng.IQService,
+              protected $state,
+              protected wpCacheService:WorkPackageCacheService) {
+
+    this.registerEventListeners();
+
+    if (this.workPackage.relations) {
+      if (!this.workPackage.relations.$loaded) {
+        this.workPackage.relations.$load().then(() => {
+          if ((this.workPackage.relations as any).count > 0) {
+            this.loadRelations();
+          }
+        });
+      } else {
+        if ((this.workPackage.relations as any).count > 0) {
+          this.loadRelations();
+        }
+      }
+    }
```

Comment:
As above, don't nest `if` and `else`. Use the logical operators instead.
PLS STAHP DOING THAT.